tests/expected/shadow/unsupported_num_objects/test.rs

```diff
@@ -15,14 +15,14 @@ fn check_max_objects<const N: usize>() {
     // - the NULL pointer whose object ID is 0, and
     // - the object ID for `i`
     while i < N {
-        let x = i;
-        assert_eq!(kani::mem::pointer_object(&x as *const usize), i + 2);
+        let x: Box<usize> = Box::new(i);
+        assert_eq!(kani::mem::pointer_object(&*x as *const usize), 2 * i + 2);
```

Comment:
What is the reason for the multiplication by 2? Is there an object for the box and another for the variable `x`? If so, would inlining the `Box:new` operation in the assert remove the doubling?